app/Commands/NewCommand.php

```diff
@@ -0,0 +1,511 @@
+    {
+        if ($this->option('editor')) {
+            $editor = $this->option('editor');
+        } else {
+            $finder = new ExecutableFinder;
+            foreach ($this->editors_gui as $_editor) {
+                if ($finder->find($_editor)) {
+                    $editor = $_editor;
+                    break;
+                }
+            }
+            if (empty($editor)) {
+                foreach ($this->editors_terminal as $_editor) {
+                    if ($finder->find($_editor)) {
+                        $editor = $_editor;
+                        break;
+                    }
+                }
+            }
+        }
+
+        if (empty($editor)) {
+            $this->warn('Unable to find a text editor to open, skipping this step.');
+
+            return false;
+        }
+
+        $this->info("Found editor $editor so am opening it now");
+
+        $process = new Process("{$editor} .");
+        $process->setWorkingDirectory($this->projectpath);
+
+        $process->setTty(true);
+        $process->run();
+
+        if (! $process->isSuccessful()) {
+            throw new ProcessFailedException($process);
+        }
+
+        return true;
+    }
+
+    protected function createDatabase($type = 'sqlite')
+    {
+        if ($type === 'sqlite') {
+            $sqlitepath = $this->projectpath . DIRECTORY_SEPARATOR . 'database' . DIRECTORY_SEPARATOR . 'database.sqlite';
+
+            File::put($sqlitepath, '');
+
+            $this->updateDotEnv([
+                'DB_CONNECTION' => 'sqlite',
+                'DB_DATABASE' => $sqlitepath,
+            ]);
+
+            return true;
+        }
+
+        if ($type === 'mysql') {
+            // @todo Offer ways to create users here, one user per db? Auto-generate passwords or always use the same?
+
+            // @todo sanitize this name
+            $sql = "CREATE database {$this->projectname}";
+
+            // @todo determine how to do this, either try root/'' or use custom config
+            //
+            $this->warn('The MySQL type is not yet implemented, sorry about that.');
+        }
+
+        return false;
+    }
+
+    protected function migrateFresh()
+    {
+        $process = new Process('php artisan migrate:fresh');
+        $process->setWorkingDirectory($this->projectpath);
+        $process->run();
+        $this->line($process->getOutput());
+    }
+
+    protected function openBrowser()
+    {
+        $this->info('Attempting to find a browser to open this in.');
+
+        $browser = '';
+        if ($this->option('browser')) {
+            $browser = $this->option('browser');
+        }
+
+        if ($this->os->isOSX()) {
+            if ($browser === '') {
+                $command = 'open "' . $this->projecturl . '"';
+            } else {
+                $command = 'open -a "' . $browser . '" "' . $this->projecturl . '"';
+            }
+        }
+
+        if ($this->os->isWindowsLike()) {
```

Comment:
Could we move this to the top and then just return ?

```php
if ($this->os->isWindowsLike()) {
    return;
}
```

not sure if there are any OSes that would not return true for either os x, windows, or unix like.. do you know?

Reply:
Assuming there can only be one of the three OS matches:

```php

        if ($this->os->isWindowsLike()) {
            $this->error("Can't open in Windows-like environments.");
            return;
        }

        if ($this->os->isOSX()) {
            if (! $this->option('browser')) {
                $command = 'open "' . $this->projecturl . '"';
            } else {
                $command = 'open -a "' . $this->option('browser') . '" "' . $this->projecturl . '"';
            }
        }

        if ($this->os->isUnixLike()) {
            if ((new ExecutableFinder)->find('xdg-open')) {
                $command = 'xdg-open "' . $this->projecturl . '"';
            }
        }

        $this->info("Opening in your browser now by executing '$command'");
        $process = new Process($command);
        $process->setWorkingDirectory($this->cwd);
        $process->run();
        $this->line($process->getOutput());
```

Reply:
Good idea, and done, although `start` was added for `isWindowsLike()` as the internet claims this to work (I'll test lambo on Windows and Linux at some point).